main.go

```diff
@@ -59,6 +60,10 @@ func fetchMirrorFreshnessMetric(backend string, url string) (float64, error) {
 		return -1, err
 	}
 
+	if resp.StatusCode != http.StatusOK {
+		return -1, errors.New("request failed with status code: " + resp.Status)
```

Comment:
totes optional: could use [fmt.Errorf](https://pkg.go.dev/errors#example-New-Errorf) instead of string concat (doesn't make any odds for this particular case, but makes it easier if you wanted to wrap the inner error for example)

Reply:
good shout!